### ISSUE TITLE
additional auto fix for parameter doesn't match name

### ISSUE THREAD

**Issue by bsutton** [dart-lang/sdk#52475]
Originally raised on the dart-code repo, Dan requested that I open the issue here.

https://github.com/Dart-Code/Dart-Code/issues/4551

The following code generates a warning:

`avoid_renaming_method_parameters`

`The parameter name 'tags' doesn't match the name 's' in the overridden method.
Try changing the name to 's'.`

```dart
  @override
  void removeTags(String tags) {
    final cmd =
        '''gcloud compute instances remove-tags $name --tags=$tags --project=$project''';

    Settings().verbose('running $cmd');
    cmd.run;
  }
```

This is because the super class's removeTags method has an argument of 's'.

The current quick fix offers to change the 'tags' argument to be 's' to match the super.

It would be nice to have an additional fix which offered to change the name of the super classes argument (e.g. from 's' to 'tags').

This does raise the question as to whether the fix should fix all other overloads to match the supers name change.

My inclination is that it should as I wouldn't be making the change if I didn't think 'tag's was the better name.

Dart version 3.0.1

**Comment by bwilkerson** [comment]
Some implementation details for anyone interested in looking at this support.

We can probably offer a fix to change the name in the superclass, but it will need to verify that the superclass is in an analysis root. We don't want to offer to make changes to files that aren't under the user's control (such as those in the pubcache).

Changing all of the other subclasses would be outside the scope of a quick fix (in terms of our implementation strategy it would be a refactoring). I'm not sure whether we want to do that or take a different approach to providing that support, so we should have a conversation before starting an implementation.

@keertip @pq
